cmd/snap/inhibit.go

```diff
@@ -45,6 +67,8 @@ func waitWhileInhibited(snapName string) error {
 	// and then either unlocks it or changes to HintInhibitedForRefresh (see
 	// gateAutoRefreshHookHandler in hooks.go).
 	// waitInhibitUnlock will return also on HintNotInhibited.
+	usedSnapDesktopIntegration := detectSnapDesktopIntegration(snapName)
+
```

Comment:
I'm confused by the location of this invocation: the comment above talks about the inhibition lock, so it should stay next to the lines below, without this invocation in the middle. Also, it's not clear why `graphicalSessionFlow()` is started after this inhibition code, but the SnapDesktopIntegration is invoked before it; I guess both should follow the same logic? If not, then this should be explained in a comment in the code.

Reply:
That's exactly what I was talking about on a previous comment: it seems that there is a bug in this logic, and the lines 72 to 78 should be deleted, because the original code does:

- if the snap is not locked, return
- wait until the snap is not locked anymore
- show a message telling the user that snap command is waiting for the snap to be unlocked
- wait until the snap is not locked anymore
- destroy the previous message
- return

So if we want to show the message while the snap is locked, we must do it before waiting for the snap to be unlocked... But also the answer was that it should go in a different MR (in fact, I tried to remove it and the tests didn't pass, they must be adapted).

Reply:
Add this bug description in the code, or open a new PR with the proposed fix and mention this PR here. Just so we don't lose this information

Reply:
Done: https://github.com/snapcore/snapd/pull/12265